elfo-dumper/src/config/dump_path.rs

```diff
@@ -0,0 +1,259 @@
+            // after = "yyy} zzz"
+            let Some((before, after)) = chunk.split_once('{') else {
+                // no more template variables.
+                plain_size += chunk.len();
+                chunk = "";
+                break;
+            };
+            plain_size += before.len();
+
+            // raw_variable = "yyy"
+            // after = " zzz"
+            let Some((raw_variable, after)) = after.split_once('}') else {
+                // 1 = {, which is not in "before".
+                plain_size += 1;
+                chunk = after;
+                continue;
+            };
+            let Some(variable) = Self::parse_variable(raw_variable) else {
+                // "{<variable>}"
+                plain_size += 2 + raw_variable.len();
+                chunk = after;
+                continue;
+            };
+
+            Self::push_path(&mut plain_size, &mut components);
+            components.push(Component {
+                // "{<variable>}"
+                size: raw_variable.len() as u16 + 2,
+                data: ComponentData::Variable(variable),
+            });
+            chunk = after;
+        }
+        if !chunk.is_empty() {
+            components.push(Component {
+                size: chunk.len() as u16,
+                data: ComponentData::Path,
+            });
+        } else {
+            Self::push_path(&mut plain_size, &mut components);
+        }
+
+        DumpPath {
+            template,
+            components,
+        }
+    }
+}
+
+impl fmt::Display for DumpPath {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.template)
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+enum ComponentData {
+    Path,
+    Variable(Variable),
+}
+
+#[derive(Debug, Clone, Copy)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+enum Variable {
+    /// Dump class.
+    Class,
+
+    /// Time of the dump.
+    Time,
+}
+
+#[derive(Debug, Clone, Copy)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+struct Component {
+    size: u16,
+    data: ComponentData,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use test_case::test_case;
+
+    #[test_case(
+        "/tmp/{class}.dump",
+        TemplateVariables {
+            class: "class",
+            ts: 0,
+        },
+        "/tmp/class.dump"
+        ; "simple class substitution"
+    )]
+    #[test_case(
+        "/tmp/dump-{class}-{time}.dump",
+        TemplateVariables {
+            class: "system.network",
+            ts: 1337100,
+        },
+        "/tmp/dump-system.network-1337100.dump"
+        ; "timestamp and class"
+    )]
+    fn it_renders_successfully(
+        template: &str,
+        vars: TemplateVariables<'_>,
+        expected: &str,
+    ) {
+        let path = DumpPath::parse(template);
+        let actual = path.render(vars);
+
+        assert_eq!(actual, expected);
+    }
+
+    #[test_case("/tmp/dump.dump", [
```

Comment:
`test_case` seems to be not very helpful here:
* extra dependency
* generates a lot of tiny tests
* the number of lines will be the same even without `test_case`

Reply:
Implemented simple DSL without the `test_case`, looks nicer and rustfmtable